File: 卡码网/Python/test19.py
#19. 单链表反转
import sys

def main():
	for line in sys.stdin:
		nums=list(map(int,line.split()))
		if nums[0]==0:
			print("list is empty")
		else:
			for i in range(1,nums[0]+1):
				if i!=1:
					print(" ",end="")
				print(nums[i],end="")
			print()
			for i in range(nums[0],0,-1):
				if i!=nums[0]:
					print(" ",end="")
				print(nums[i],end="")
			print()

if __name__ == '__main__':
	main()

File: 卡码/KamaCoder/Js/test19.js
//19. 单链表反转
const rl=require("readline").createInterface({input:process.stdin});
const iter=rl[Symbol.asyncIterator]();
const readline=async ()=>(await iter.next()).value;

const out=process.stdout;

async function main(){
  while(line=await readline()){
    const nums=line.split(" ").map(Number);
    if(nums[0]===0) console.log("list is empty");
    else{
      let res="";
      for(let i=1;i<=nums[0];i++){
        if(i!=1) res+=" ";
        res+=nums[i];
      }
      console.log(res);
      res="";
      for(let i=nums[0];i>=1;i--){
        res+=nums[i];
        if(i!=1) res+=" ";
      }
      console.log(res);
    }
  }
}

main();